core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -810,6 +812,22 @@ static void loadFileMetadataInternal(RpcContext rpcContext, LockedInodePath inod
       // This may occur if a thread created or loaded the file before we got the write lock.
       // The file already exists, so nothing needs to be loaded.
       LOG.debug("Failed to load file metadata: {}", e.toString());
+    } catch (Throwable e) {
+      // Clean up possibly incomplete files due to any exceptions
+      try (LockedInodePath writeLockedPath = inodePath.lockFinalEdgeWrite()) {
+        FileInfo info = fsMaster.getFileInfoInternal(writeLockedPath);
+        if (!info.isCompleted()) {
+          fsMaster.deleteInternal(rpcContext, inodePath, DeleteContext
+              .mergeFrom(DeletePOptions.newBuilder().setAlluxioOnly(true)));
+        }
+      } catch (FileDoesNotExistException ex) {
+        LOG.info("File not created yet, do not need to clean up: {}", ex.toString());
+      } catch (DirectoryNotEmptyException ex) {
+        LOG.warn("Trying to remove a directory, expecting a file: {}", ex.toString());
```

Comment:
I am also wondering whether these should be suppressed. I guess at least there is logging, but I guess it's not the worst if we don't since we're at least logging them